src/autorest-core/app.ts

```diff
@@ -469,26 +460,22 @@ async function main() {
       exitcode = await currentMain(autorestArgs);
     }
 
-    // for relaxed profiling (assuming that no one calls `main` from electron... use AAAL!)
-    if (require("process").versions.electron) await new Promise(_ => { });
-
     process.exit(exitcode);
   } catch (e) {
+    // be very careful about the following check:
+    // - doing the inversion (instanceof Error) doesn't reliably work since that seems to return false on Errors marshalled from safeEval
     if (e instanceof Exception) {
-      console.log(e.message);
-
-      if (autorestArgs.indexOf("--debug")) {
+      if (autorestArgs.indexOf("--debug") !== -1) {
         console.log(e);
+      } else {
+        console.log(e.message);
       }
       process.exit(e.exitCode);
     }
 
-    if (e instanceof Error) {
+    if (e !== false) {
```

Comment:
if `Process` doesn't return `true`, we throw what was returned (line 277ish)... not so cool to print that in case its `false`